sources/System/Collision/CollisionSystem.cpp

```diff
@@ -0,0 +1,52 @@
+//
+// Created by Louis Auzuret on 5/20/21
+//
+
+#include "Component/Movable/MovableComponent.hpp"
+#include "Component/Position/PositionComponent.hpp"
+#include "Component/Collision/CollisionComponent.hpp"
+#include "System/Collision/CollisionSystem.hpp"
+
+namespace BBM
+{
+	CollisionSystem::CollisionSystem(WAL::Wal &wal)
+	: WAL::System({typeid(PositionComponent), typeid(CollisionComponent)}), _wal(wal)
+	{
+
+	}
+
+	bool CollisionSystem::collide(Vector3f minA, Vector3f maxA, Vector3f minB, Vector3f maxB)
+	{
+		bool overlapX = (minA.x <= maxB.x && maxA.x >= minB.x) || (minB.x <= maxA.x && maxB.x >= minA.x);
+		bool overlapY = (minA.y <= maxB.y && maxA.y >= minB.y) || (minB.y <= maxA.y && maxB.y >= minA.y);
+		bool overlapZ = (minA.z <= maxB.z && maxA.z >= minB.z) || (minB.z <= maxA.z && maxB.z >= minA.z);
+
+		return (overlapX && overlapY && overlapZ);
+	}
+
+	void CollisionSystem::onFixedUpdate(WAL::Entity &entity)
+	{
+		auto &posA = entity.getComponent<PositionComponent>();
+		auto &col = entity.getComponent<CollisionComponent>();
+		Vector3f position = posA.position;
+		if (entity.hasComponent(typeid(MovableComponent)))
+			position += entity.getComponent<MovableComponent>().getVelocity();
+		Vector3f minA = Vector3f::min(position, position + col.bound);
+		Vector3f maxA = Vector3f::max(position, position + col.bound);
+		for (auto &other : _wal.scene->getEntities()) {
+			if (&other == &entity)
+				continue;
+			if (!other.hasComponent(typeid(CollisionComponent)) ||
```

Comment:
Same as before, use the templated version

Reply:
fixed -> d0205b6